studiopress/atomic-blocks
# Redirect to the Migrate page on activation and upgrade

On plugin activation and version upgrade, this redirects to the Migrate page.

Fixes [GF-2675](https://wpengine.atlassian.net/browse/GF-2675)


### How to test
It's not possible to test updating Atomic Blocks, as that would of course remove this code.

So this tests updating Akismet:

1. Install [Akismet](https://wordpress.org/plugins/akismet/)
2. Edit the `Version` in `akismet.php` to be outdated: `4.2.0`:

```php
<?php
/**
 * @package Akismet
 */
/*
Plugin Name: Akismet Anti-Spam
Plugin URI: https://akismet.com/
Description: Used by millions, Akismet is quite possibly the best way in the world to <strong>protect your blog from spam</strong>. It keeps your site protected even while you sleep. To get started: activate the Akismet plugin and then go to your Akismet Settings page to set up your API key.
Version: 4.2.0
```
3. Edit Atomic Blocks to look for Akismet, not Atomic Blocks:
```diff
diff --git a/dist/migration/class-redirect.php b/dist/migration/class-redirect.php
index df2eafd..4af8318 100644
--- a/dist/migration/class-redirect.php
+++ b/dist/migration/class-redirect.php
@@ -32,7 +32,7 @@ class Redirect {
         */
        public function redirect_after_upgrade( $upgrader, $extra ) {
                unset( $upgrader );
-               if ( current_user_can( Notice::CAPABILITY ) && isset( $extra['plugins'] ) && in_array( 'atomic-blocks/atomicblocks.php', $extra['plugins'], true ) ) {
+               if ( current_user_can( Notice::CAPABILITY ) && isset( $extra['plugins'] ) && in_array( 'akismet/akismet.php', $extra['plugins'], true ) ) {
                        add_option( 'atomic_blocks_do_activation_redirect', true );
                }
```

4. Go to `/wp-admin/plugins.php`
5. You should see that Akismet needs an update: ![Screen Shot 2022-01-10 at 1 51 01 PM](https://user-images.githubusercontent.com/4063887/148829953-bd1693c9-99e8-4e17-85c8-2b81b635133c.png)

6. If you don't see that Akismet needs an update, go to `/wp-admin/update-core.php` and click 'Check again'
7. Update Akismet
8. The update happens via AJAX, so nothing should happen yet, other than the update succeeding
9. Reload the page
10. Expected, it's redirected to `/wp-admin/admin.php?page=atomic-blocks-migrate-page`: ![Screen Shot 2022-01-10 at 1 53 41 PM](https://user-images.githubusercontent.com/4063887/148830258-338835cc-2103-4bce-9c91-d32371fad425.png)



### Documentation
No documentation required. 



### Suggested changelog entry

- Redirect to Migration page on activation and upgrade

### Notes
* It looks like upgrading plugins on `/wp-admin/plugins.php` happens via AJAX.
* So simply doing `wp_safe_redirect()` won't redirect, it'll make the AJAX call fail.
* Instead, this uses the existing logic to redirect on the next page load.
